website/content/docs/api-clients/api/pagination.mdx

```diff
@@ -0,0 +1,129 @@
+- `sort_by` - The column name that the returned items should be sorted on.
+Each list endpoint has one defined column which defines the sorting for all returned items.
+This attribute is included in the response, so that the sort order for each endpoint is explicit.
+- `sort_dir` - The direction in which the returned items are sorted.
+Responses can include `desc` for descending, or `asc` for ascending.
+- `items` - The list of items.
+The items represent replacements for any existing items with matching IDs in the client's cache.
+- `est_item_count` - An unsigned integer that indicates the total number of items that are available in the list of results.
+This number may be an estimate, since the number of items may change during the pagination process.
+
+### Example response
+
+```json
+{
+  "response_type": "<delta|complete>",
+  "pagination_token": "<opaque-token>",
+  "sort_by": "created_time",
+  "sort_dir": "desc",
+  "est_item_count": 1000,
+  "items": [
+    {
+      "id": "ttcp_1234567890",
+      "scope_id": "p_1234567890"
+    },
+    {
+      "id": "ttcp_2234567890",
+      "scope_id": "p_1234567890"
+    }
+  ]
+}
+```
+
+## List tokens
+
+List tokens contain general metadata associated with the ongoing pagination and either a pagination or refresh token, depending on the phase of the pagination.
+If you compose an invalid token, you receive an invalid token error.
+
+When you make a `list` request without a `list_token`, Boundary builds a page of items that contains all the resources currently available, subject to the page size, with the following attributes:
+
+- Must contain a `response_type` equal to `delta` or `complete`
+- Must contain a `sort_by`
+- Must contain a `sort_dir`
+- May contain a `list_token`
+- May contain `items`
+- May contain `est_item_count`
+
+When you make a `list` request using a valid `list_token`, Boundary builds a page of items following from the page returned with the token:
+
+- Must contain a `response_type` equal to `delta` or `complete`
+- Must contain a `sort_by`
+- Must contain a `sort_dir`
+- Must contain a `list_token`
+- May contain `items`
+- May contain `est_item_count`
+
+If you make a `list` request using a token that has a `create_time` older than 30 days, Boundary treats it as an invalid token and returns an error.
+
+## Constraints and limitations
+
+In scenarios where updates are so frequent that the results may change while the refresh operation runs, those updates may not appear at all in the refreshed list.
+If you make frequent updates to a resource, or if you are updating data while you run a `list` operation, HashiCorp recommends that you generate the full list of results instead of performing a refresh.
+
+Workers are not included in refresh operations because the updates they send to the controller are too frequent to be captured in a refresh.
+To update a list of worker resources, you must regenerate the full list of results.
```

Comment:
This is great. Can we also add a notice that there may be duplicate items returned in the "refresh phase" pages, and that clients need to be able to handle these. This is a fundamental tradeoff when working across multiple database transactions, with the alternative being that sometimes items are missed from the pagination.

Reply:
Sure! I added this:

>When you refresh the list of results, you may notice duplicate items in the list.
This is a fundamental tradeoff when you search across multiple database transactions.

Should we provide any additional follow up advice here? Or is it just something for them to be aware of?

Reply:
I think this is fine, I just want it to be clear that it is the responsibility of the user to handle duplicates in the responses.